src/main/java/com/eprosima/fastdds/idl/templates/TypeObjectTestingTestSource.stg

```diff
@@ -0,0 +1,1101 @@
+
+    $get_type_object(ctx=ctx, type=bitmask, relation=relation, relation_name=relation_name, var_name=["typeobject"])$
+    ASSERT_EQ(eprosima::fastrtps::types::EK_$if(relation)$COMPLETE$else$MINIMAL$endif$, $bitmask.name$_$relation_name$_typeobject->_d());
+
+    $relation_capname$TypeObject $bitmask.name$_$relation_name$_object = $bitmask.name$_$relation_name$_typeobject->$relation_name$();
+    ASSERT_EQ(TK_BITMASK, $bitmask.name$_$relation_name$_object._d());
+
+    $relation_capname$BitmaskType $bitmask.name$_$relation_name$_bitmask = $bitmask.name$_$relation_name$_object.bitmask_type();
+    //BitmaskTypeFlag flags;// Unused. No flags apply
+
+    $relation_capname$BitmaskHeader $bitmask.name$_$relation_name$_header = $bitmask.name$_$relation_name$_bitmask.header();
+    CommonBitmaskHeader $bitmask.name$_$relation_name$_commonheader = $bitmask.name$_$relation_name$_header.common();
+
+    BitBound $bitmask.name$_$relation_name$_commonheader_bit_bound = $bitmask.name$_$relation_name$_commonheader.bit_bound();
+    ASSERT_EQ($bitmask.bitBound$, $bitmask.name$_$relation_name$_commonheader_bit_bound);
+
+    $relation_capname$BitflagSeq $bitmask.name$_$relation_name$_flagseq = $bitmask.name$_$relation_name$_bitmask.flag_seq();
+>>
+
+check_bitmask_flag_type(ctx, flag, relation, relation_name, relation_capname) ::= <<
+    //=====check_bitmask_flag_type - $relation_name$ - $flag.name$=====
+    $relation_capname$Bitflag $bitmask.name$_$relation_name$_$flag.name$ = $bitmask.name$_$relation_name$_flagseq.at(flagId);
+    CommonBitflag $bitmask.name$_$relation_name$_common_$flag.name$_bitmask_flag = $bitmask.name$_$relation_name$_$flag.name$.common();
+    
+    unsigned short $bitmask.name$_$relation_name$_common_$flag.name$_bitmask_flag_position = $bitmask.name$_$relation_name$_common_$flag.name$_bitmask_flag.position();
+    ASSERT_EQ(flagId, $bitmask.name$_$relation_name$_common_$flag.name$_bitmask_flag_position);
+
+    //BitflagFlag flags;// Unused. No flags apply
+
+    $if(relation)$
+    {
+    CompleteMemberDetail $flag.name$_$relation_name$_detail = $bitmask.name$_$relation_name$_$flag.name$.detail();
+    $check_optional_detail_annotations(ctx=ctx, object=flag)$
+    MemberName $flag.name$_$relation_name$_detail_flag_name = $flag.name$_$relation_name$_detail.name();
+    ASSERT_EQ("$flag.name$", $flag.name$_$relation_name$_detail_flag_name);
+    }
+    $else$
+    {
+    MinimalMemberDetail $bitmask.name$_$relation_name$_detail = $bitmask.name$_$relation_name$_$flag.name$.detail();
+    NameHash $flag.name$_$relation_name$_flag_name_hash = $bitmask.name$_$relation_name$_detail.name_hash();
+    MD5 $flag.name$_$relation_name$_name_hash("$flag.name$");
+    for(int i = 0;i < 4;++i)
+    {
+        ASSERT_EQ($flag.name$_$relation_name$_name_hash.digest[i], $flag.name$_$relation_name$_flag_name_hash[i]);
+    }
+    }
+    $endif$
+
+    flagId++;
```

Comment:
The flag positions do not have to be sequential. For instance, the [test IDL](https://github.com/eProsima/dds-types-test/blob/main/IDL/helpers/basic_inner_types.idl#L17)